backend/student-internship/src/main/java/com/example/studentinternship/controller/CompanyController.java

```diff
@@ -21,32 +23,40 @@ public CompanyController(CompanyService companyService)
     }
 
     @PostMapping
-    public ResponseEntity<Company> createCompany(@RequestBody Company company)
+    public ResponseEntity<CompanyDto> createCompany(@RequestBody CompanyDto company)
     {
-        Company newCompany = companyService.createCompany(company);
+        CompanyDto newCompany = companyService.createCompany(company);
         return new ResponseEntity<>(newCompany, HttpStatus.CREATED);
     }
 
     @GetMapping
-    public ResponseEntity<List<Company>> getAllCompanies()
+    public ResponseEntity<List<CompanyDto>> getAllCompanies()
     {
-        List<Company> companies = companyService.getAllCompanies();
+        List<CompanyDto> companies = companyService.getAllCompanies();
         return ResponseEntity.ok().body(companies);
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Company> getCompanyById(@PathVariable(value = "id") String companyId)
+    public ResponseEntity<CompanyDto> getCompanyById(@PathVariable(value = "id") String companyId)
     {
-        Company company = companyService.getCompanyById(companyId)
+        CompanyDto company = companyService.getCompanyById(companyId)
+                .orElseThrow(() -> new CompanyNotFoundException(companyId));
+        return ResponseEntity.ok().body(company);
+    }
+
+    @GetMapping("/user/{userId}")
+    public ResponseEntity<CompanyDto> getCompanyByUserId(@PathVariable(value = "userId") String companyId)
```

Comment:
Aici cred ca voia sa fie userId in loc de companyId, nu?

Reply:
Ups, asa e. L-am modificat